### PR TITLE
Add Dockerfile for training.

### PR DESCRIPTION
**Pull request by tushuhei**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,62 @@ yarn build
 
 This will minify SASS and JS for serving in production.
 
+## Build your own model
+You can build your own image recognition model by running a Docker container.
+
+Before you start training, you'll need a set of images to teach the image
+recognition model about the labels you want to recognize.
+Prepare images for trainings and divide them into directories of the label
```
Comment (jbruwer) [inline README.md:39]: Maybe this line will read a little easier as follows

"Prepare images for training by dividing them into directories for each label name that you want to train" Does that still sound correct?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+names.
+If you are about to train a recognition model which classifies cat and dog,
```
Comment (jbruwer) [inline README.md:41]: For example: the directory structure for training **cat** and **dog** will look as follows assuming image data is stored under `data/images`

File: README.md [hunk 3]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+They are a SavedModel in a web-friendly format converted by
+[TensorFlow.js converter](https://github.com/tensorflow/tfjs-converter).
+You can build your own game on your custom image recognition model by replacing
+correspondings files under `dist/model/` directory with them.
+You also need to update `src/js/scavenger_classes.ts` in order to label outputs
+from the custom model with human-readable strings.
+Update the game logic in `src/js/game.ts` if needed.
+
+### Using GPU
+You can boost training speed by utilizing GPU in general.
```
Comment (jbruwer) [inline README.md:82]: You can boost the training speed by utilizing your GPU.

File: README.md [hunk 4]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+    │   ├── cat1.jpg
+    │   ├── cat2.jpg
+    │   └── ...
+    └── dog
+        ├── dog1.jpg
+        ├── dog2.jpg
+        └── ...
+```
+
+Once the training images are ready, you can kickstart the training by building
```
Comment (jbruwer) [inline README.md:58]: Does it still work to say

"Once the sample images are ready, you can kickstart the training by building and running the Docker container"

File: README.md [hunk 5]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+[TensorFlow.js converter](https://github.com/tensorflow/tfjs-converter).
+You can build your own game on your custom image recognition model by replacing
+correspondings files under `dist/model/` directory with them.
+You also need to update `src/js/scavenger_classes.ts` in order to label outputs
+from the custom model with human-readable strings.
+Update the game logic in `src/js/game.ts` if needed.
+
+### Using GPU
+You can boost training speed by utilizing GPU in general.
+If you want to use GPU for training, install
```
Comment (jbruwer) [inline README.md:83]: use **the** GPU

File: README.md [hunk 6]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+After the training is completed, you'll see three files in
+`data/saved_model_web` directory:
+
+- web_model.pb (the dataflow graph)
+- weights_manifest.json (weight manifest file)
+- group1-shard\*of\* (collection of binary weight files)
+
+They are a SavedModel in a web-friendly format converted by
+[TensorFlow.js converter](https://github.com/tensorflow/tfjs-converter).
+You can build your own game on your custom image recognition model by replacing
```
Comment (jbruwer) [inline README.md:75]: by replacing **the**

File: README.md [hunk 7]
```diff
@@ -31,6 +31,62 @@ yarn build
...
+correspondings files under `dist/model/` directory with them.
+You also need to update `src/js/scavenger_classes.ts` in order to label outputs
```
Comment (jbruwer) [inline README.md:77]: in order to update the label outputs

**Review by jbruwer — COMMENTED** [review]
See comment suggestions. Thanks!

**Comment by tushuhei** [comment]
Thank you for your review! Let me restart this change in another CR #2 generated from the branch `add-dockerfile` branch.

**Comment by tushuhei** [comment]
Also, your comments are reflected in the new CR. I wanted to continue to use this PR, but I couldn't squash my commits nicely.. Thanks!

**Pull request closed without merging**